satpy/readers/amsr2_l2_gaasp.py

```diff
@@ -0,0 +1,286 @@
+southern hemisphere or both depending on what files are provided to Satpy, this
+reader appends a `_NH` and `_SH` suffix to all variable names that are
+dynamically discovered from the provided files.
+
+"""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+from pyproj import CRS
+from pyresample.geometry import AreaDefinition
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy import CHUNK_SIZE
+
+logger = logging.getLogger(__name__)
+
+
+class GAASPFileHandler(BaseFileHandler):
+    """Generic file handler for GAASP output files."""
+
+    y_dims = (
+        'Number_of_Scans',
+    )
+    x_dims = (
+        'Number_of_hi_rez_FOVs',
+        'Number_of_low_rez_FOVs',
+    )
+    time_dims = (
+        'Time_Dimension',
+    )
+    is_gridded = False
+    dim_resolutions = {
+        'Number_of_hi_rez_FOVs': 5000,
+        'Number_of_low_rez_FOVs': 10000,
+    }
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Open the NetCDF file with xarray and prepare the Dataset for reading."""
+        super().__init__(filename, filename_info, filetype_info)
+        chunks = {dim_name: CHUNK_SIZE for dim_name in
+                  self.y_dims + self.x_dims + self.time_dims}
+        self.nc = xr.open_dataset(self.filename,
+                                  decode_cf=True,
+                                  mask_and_scale=False,
+                                  chunks=chunks)
+
+        if len(self.time_dims) == 1:
+            self.nc = self.nc.rename({self.time_dims[0]: 'time'})
+
+    @property
+    def start_time(self):
+        """Get start time of observation."""
+        time_str = self.nc.attrs['time_coverage_start']
+        return datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    @property
+    def end_time(self):
+        """Get end time of observation."""
+        time_str = self.nc.attrs['time_coverage_end']
+        return datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    @property
+    def sensor_names(self):
+        """Sensors who have data in this file."""
+        return {self.nc.attrs['instrument_name'].lower()}
+
+    @property
+    def platform_name(self):
+        """Name of the platform whose data is stored in this file."""
+        return self.nc.attrs['platform_name']
+
+    def _get_var_name_without_suffix(self, var_name):
+        var_suffix = self.filetype_info.get('var_suffix', "")
+        if var_suffix:
+            var_name = var_name[:-len(var_suffix)]
+        return var_name
+
+    def _scale_data(self, data_arr, attrs):
+        # handle scaling
+        # take special care for integer/category fields
+        scale_factor = attrs.pop('scale_factor', 1.)
+        add_offset = attrs.pop('add_offset', 0.)
+        scaling_needed = not (scale_factor == 1 and add_offset == 0)
+        if scaling_needed:
+            data_arr = data_arr * scale_factor + add_offset
```

Comment:
Question here: what is the data type? Just checking if the data shouldn't be in float32's for example (as `1.` and `0.` are float64s)

Reply:
Good point. I will double check.

Reply:
Just checked, the data is indeed "float" from ncdump which the internet tells me is 32-bit (double is 64-bit).

Reply:
As mentioned on slack, with my version of numpy/xarray/dask it seems that multiplying a python float on a 32-bit float array keeps it as a 32-bit float.